adaptive/tests/test_sequence_learner.py

```diff
@@ -0,0 +1,28 @@
+import asyncio
+
+import numpy as np
+
+from adaptive import Runner, SequenceLearner
+from adaptive.runner import SequentialExecutor
+
+
+class FailOnce:
+    def __init__(self):
+        self.failed = False
+
+    def __call__(self, value):
+        if self.failed:
+            return value
+        self.failed = True
+        raise Exception
```

Comment:
```suggestion
        raise RuntimeError
```
is more appropriate even in a mock.